### PR TITLE
ov 2.0: python image_retrieval_demo and place_recognition_demo

### PR DESCRIPTION
**Pull request by anzhella-pankratova**
Migrate image_retrieval_demo and place_recognition_demo to OV2.0

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: demos/image_retrieval_demo/python/image_retrieval_demo/image_retrieval.py [hunk 1]
```diff
@@ -32,23 +32,24 @@ class IEModel(): # pylint: disable=too-few-public-methods
...
-            ie.add_extension(cpu_extension, 'CPU')
+            core.add_extension(cpu_extension, 'CPU')
 
         path = '.'.join(model_path.split('.')[:-1])
         log.info('Reading model {}'.format(model_path))
-        self.net = ie.read_network(path + '.xml', path + '.bin')
-        self.output_name = list(self.net.outputs.keys())[0]
-        self.exec_net = ie.load_network(network=self.net, device_name=device)
+        self.model = core.read_model(path + '.xml', path + '.bin')
+        self.input_tensor_name = "Placeholder"
```
Comment (eaidova) [inline demos/image_retrieval_demo/python/image_retrieval_demo/image_retrieval.py:42]: why we can not get input_tensor_name from model here?

File: demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py [hunk 2]
```diff
@@ -31,24 +31,24 @@ class IEModel: # pylint: disable=too-few-public-methods
...
-            ie.add_extension(cpu_extension, 'CPU')
+            core.add_extension(cpu_extension, 'CPU')
 
         log.info('Reading model {}'.format(model_path))
-        self.net = ie.read_network(model_path, model_path.with_suffix('.bin'))
-        self.input_name = next(iter(self.net.input_info))
-        self.output_name = next(iter(self.net.outputs))
-        self.input_size = self.net.input_info[self.input_name].input_data.shape
-        self.exec_net = ie.load_network(network=self.net, device_name=device)
+        self.model = core.read_model(model_path, model_path.with_suffix('.bin'))
```
Comment (eaidova) [inline demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py:39]: can we unify model reading (in some demos, only model provided, in some demos 2 files using with_suffix, below also model_path[:-4] + '.xml'/',bin' used?

File: demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py [hunk 3]
```diff
@@ -31,24 +31,24 @@ class IEModel: # pylint: disable=too-few-public-methods
...
+        self.input_tensor_name = self.model.inputs[0].get_any_name()
+        self.input_size = list(self.model.input(self.input_tensor_name).shape)
```
Comment (eaidova) [inline demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py:41]: is it necessary to wrap shape to list?

File: demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py (hunk 2 above)
Comment (anzhella-pankratova) [reply demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py:39]: Yeah, I will also unify it in the other PR once the all py demos will be migrated to ov2.0 and merged. Are you okay with it?

File: demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py (hunk 3 above)
Comment (anzhella-pankratova) [reply demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py]: We cannot apply slicing to tensor shape as in the old API, so the solution here is to wrap to list. Replaced by getting the width and height by indices.

File: demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py (hunk 2 above)
Comment (eaidova) [reply demos/place_recognition_demo/python/place_recognition_demo/place_recognition.py:39]: yes, it is not stopper, just common recommendation :)

**Review by eaidova — APPROVED** [review]

**Pull request merged**